0xProject/0x-launch-kit-frontend
# Add row for total ETH balance

Closes #120.

## How to test

- A row with the *total* ETH balance (ETH + wETH) of the connected user should be the first row in the token balances component.
- Wrapping/unwrapping ether shouldn't change the balance of this row (actually it should change a little for the gas used, but with the formatting of the balance this shouldn't be noticed most of the time).
- The lock/unlock button is used to lock/unlock the wETH token. There's no concept of locking/unlocking for plain ETH.

## Implementation

Before this, we were saving the wETH *balance* (a `BigNumber`) in the store. Now we have a `wethTokenBalance` field of type `TokenBalance`. This is needed to keep track of the `isUnlocked` state of the wETH token.

An alternative for this was adding the wETH token to the `tokenBalances` array. I tried this, but it turned out to be messy. This new row needs some special handling (the label isn't just the name of the token, and the balance isn't just the token balance), so having it in the same list as the other tokens wasn't worth it.

When the app starts, we don't know the value of this `wethTokenBalance`, so I made it nullable. This adds some boilerplate here and there, but it's manageable.

You'll notice that I unified the `lockToken` and `unlockToken` actions into one: `toggleTokenLock`. These two functions already had some repetition, and adding the special handling of the wETH token was going to make it even less DRY. Now everything is handled in that action.

## Screenshot

![image](https://user-images.githubusercontent.com/417134/54156780-b3a62680-4425-11e9-9c66-446bd882f1a1.png)
